### PR TITLE
Added ExecutablePool documentation

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 1]
```diff
@@ -121,6 +121,36 @@ Setting fields:
...
+
+That dictionary source can be configured only via XML configuration. Creating dictionaries with executable source via DDL is disabled, otherwise, the DB user would be able to execute arbitrary binary on ClickHouse node.
+
+## Executable Pool {#dicts-external_dicts_dict_sources-executable_pool}
+
+Executable pool allows loading data from pool of processes. This source does not work with dictionary layouts that need to load all data from source. Executable pool works if the dictionary is stored using `cache`, `ssd_cache`, `direct` layouts. Executable pool will spawn pool of processes with specified command and keep them running until they exit. The program should read data from STDIN while it is available and output result to STDOUT, and it can wait for next block of data on stdin. ClickHouse will not close STDIN after processing a block of data but will pipe another chunk of data when needed. The executable script should be ready for this way of data processing - it should poll STDIN and flush data to STDOUT early.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:130]: complex_key_cache, complex_key_ssd_cache, complex_key_direct

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md [hunk 2]
```diff
@@ -121,6 +121,36 @@ Setting fields:
...
+
+Example of settings:
+
+``` xml
+<source>
+    <executable_pool>
+        <command>cat /opt/dictionaries/os.tsv</command>
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md:137]: But this is not suitable for pool.

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md (hunk 1 above)
Comment (kitaisreal) [reply docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md]: Fixed.

File: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md (hunk 2 above)
Comment (kitaisreal) [reply docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources.md]: Fixed.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by gyuton** [comment]
Internal documentation ticket: DOCSUP-10300.

**Pull request merged**